coalib/results/TextPosition.py

```diff
@@ -7,20 +7,35 @@
 class TextPosition:
 
     @enforce_signature
-    def __init__(self, line: (int, None)=None, column: (int, None)=None):
+    def __init__(self,
+                 line: (int, None)=None,
+                 column: (int, None)=None,
+                 text: (tuple, list, None)=None,
```

Comment:
FWIW you can drive an object from text position that holds this info in
addition
On 27 Mar 2016 12:07 p.m., "Abhay Raizada" <notifications@github.com> wrote:

> In coalib/results/TextPosition.py
> <https://github.com/coala-analyzer/coala/pull/1947#discussion_r57526212>:
>
> > @@ -7,20 +7,35 @@
> >  class TextPosition:
> >
> >      @enforce_signature
> > -    def __init__(self, line: (int, None)=None, column: (int, None)=None):
> > +    def __init__(self,
> > +                 line: (int, None)=None,
> > +                 column: (int, None)=None,
> > +                 text: (tuple, list, None)=None,
>
> hmm, so how about we never store text? just have conversion functions? ,
> assert that line, col and position belong to the same character during
> instantiation? but then we wont be able to calculate position, line and
> column, lazily, and the text would be parsed everytime?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/1947/files/42292b78c205f682983b4a5a7e091dded456a040#r57526212>
>